code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -978,6 +988,32 @@
 	set category = "Robot Commands"
 	set_autosay()
 
+/mob/living/silicon/robot/proc/control_headlamp()
+	if(stat)
+		src << "<span class='danger'>This function is currently offline.</span>"
+		return
+
+//Some sort of magical "modulo" thing which somehow increments lamp power by 2, until it hits the max and resets to 0.
+	lamp_intensity = (lamp_intensity+2) % (lamp_max+2)

```

Comment:
and if you didn't know, for learning purposes:

```
0+2 % 12 = 2
2+2 % 12 = 4
4+2 % 12 = 6
6+2 % 12 = 8
8+2 % 12 = 10
10+2 % 12 = 0
```
is how your current code runs, which is what you wanted.
x mod y is x so long as x is not y, or 
```
var/result = 0
if(x != y) 
    result = x
else
    result = 0
```